source/evp_override.c

```diff
@@ -706,6 +707,35 @@ const EVP_MD *EVP_sha512() {
     return &md;
 }
 
+EVP_MD *EVP_MD_fetch(OSSL_LIB_CTX *ctx, const char *algorithm, const char *properties) {
+    // Assume this method is always called without a library context
+    assert(ctx != NULL);
+
+    // Reuse the old initialization values to set value
+    EVP_MD *md_value = NULL;
+    if (strcmp(algorithm, "MD5") == 0) {
+        md_value = EVP_md5();
+    } else if (strcmp(algorithm, "SHA1") == 0) {
+        md_value = EVP_sha1();
+    } else if (strcmp(algorithm, "SHA224") == 0) {
+        md_value = EVP_sha224();
+    } else if (strcmp(algorithm, "SHA256") == 0) {
+        md_value = EVP_sha256();
+    } else if (strcmp(algorithm, "SHA384") == 0) {
+        md_value = EVP_sha256();
+    } else if (strcmp(algorithm, "SHA512") == 0) {
+        md_value = EVP_sha512();
+    }
+
+    if (md_value) {
+        EVP_MD *md = malloc(sizeof(EVP_MD));
```

Comment:
Assuming that the EVP_shaXXX() functions above do some sort of "malloc()" call, then we "malloc()" again, copy the value, and return. Is it possible to do this with just one malloc() instead of two?

Reply:
The EVP_shaXXX functions don't do a malloc call. They return `const EVP_MD`, so they just keep returning references to the same static EVP_MD struct:
https://github.com/awslabs/aws-verification-model-for-libcrypto/blob/efb3924128cb1d2c977cc6d8203a1e77181c0bc6/source/evp_override.c#L689-L692

That's why I'm doing it this way. I do one malloc, then use the const values from the old methods to set the value of that memory.

Reply:
Understood.